cpp/src/decisiontree/decisiontree.h

```diff
@@ -81,75 +81,106 @@ struct DecisionTreeParams {
 	 */
 	int min_rows_per_node = 2;
 	/**
-	 * Wheather to bootstarp columns with or without replacement
+	 * Whether to bootstrap columns with or without replacement.
 	 */
 	bool bootstrap_features =  false;
-	
+	/**
+	 * Node split criterion. GINI and Entropy for classification, MSE or MAE for regression.
+	 */
+	CRITERION split_criterion = CRITERION_END;
+
 	DecisionTreeParams();
-	DecisionTreeParams(int cfg_max_depth, int cfg_max_leaves, float cfg_max_features, int cfg_n_bins, int cfg_split_aglo, int cfg_min_rows_per_node, bool cfg_bootstrap_features);
+	DecisionTreeParams(int cfg_max_depth, int cfg_max_leaves, float cfg_max_features, int cfg_n_bins, int cfg_split_aglo, int cfg_min_rows_per_node, bool cfg_bootstrap_features, CRITERION cfg_split_criterion);
 	void validity_check() const;
 	void print() const;
 };
 
-template<class T>
-class DecisionTreeClassifier {
+template<class T, class L>
+class dt {
```

Comment:
Maybe better to name this as `DecisionTreeBase`?

Reply:
agreed. Will change the base class name.

Reply:
Base class name changed. Thanks @myrtw